lib/floe/workflow.rb

```diff
@@ -51,11 +50,11 @@ def step
 
       context.state["FinishedTime"] = Time.now.utc
       context.state["Duration"]     = tock - tick
-      context.state["Output"]       = output
-      context.state["NextState"]    = next_state
-      context.state["Error"]        = current_state.error if current_state.respond_to?(:error)
-      context.state["Cause"]        = current_state.cause if current_state.respond_to?(:cause)
-      context.execution["EndTime"]  = Time.now.utc if next_state.nil?
+      if current_state.respond_to?(:error)
+        context.end_state!(output, :error => current_state.error, :cause => current_state.cause)
+      else
+        context.end_state!(output, next_state)
+      end
```

Comment:
What if we move this into `State` rather than `Context`?  I think the fact that we're checking `State#respond_to?(:error)` is an indication that this is in the wrong place.

When I think of setting initial `context.state` values it sounds like something that is done by `State#run*` or `State#before_start`

Reply:
yea. I want to move there
There reason why I was ok with `respond_to?` is because we are moving that call into `State` in the short term. (either your PR or mine)